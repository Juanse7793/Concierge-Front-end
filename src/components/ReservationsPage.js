import { useDispatch, useSelector } from 'react-redux';
import { deleteReservation } from '../redux/reducers/reservations';
import Sidebar from './Sidebar';

const ReservationsPage = () => {
  const user = useSelector((state) => state.user.user);
  const events = useSelector((state) => state.events.events);
  const reservations = useSelector((state) => state.reservations.reservations.filter(
    (reservation) => reservation.user_id === user.id,
  ));
  const dispatch = useDispatch();

  const deleteReservationHandler = (e) => {
    dispatch(deleteReservation(Number(e.target.id)));
  };

  return (
    <div className="row">
      <Sidebar />
      <section>
        {reservations.map((reservation) => (
          <ul key={reservation.id}>
            <li className="row reservation-name">
              <p>
                {events.find((event) => event.id === reservation.event_id).name}
              </p>
              <button
                type="button"
                className="delete-btn"
                id={reservation.id}
                onClick={deleteReservationHandler}
              >
                Delete
              </button>
            </li>
          </ul>
        ))}
      </section>
    </div>
  );
};

export default ReservationsPage;
