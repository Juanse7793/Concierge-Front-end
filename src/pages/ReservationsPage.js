import { useDispatch, useSelector } from 'react-redux';
import { deleteReservation } from '../redux/reducers/users';
import ListItem from '../components/ListItem';

const ReservationsPage = () => {
  const user = useSelector((state) => state.user.user);
  const events = useSelector((state) => state.events.events);
  const dispatch = useDispatch();

  const deleteReservationHandler = (e) => {
    dispatch(deleteReservation(Number(user.id), Number(e.target.id)));
  };

  return (
    <section>
      {user.reservations.map((reservation) => (
        <ul key={reservation.id}>
          {events.length <= 0 ? (<li className="row">Please wait...</li>) : (
            <ListItem
              name={events.find((event) => event.id === reservation.event_id).name}
              start={reservation.start_date}
              end={reservation.end_date}
              deleteFunc={deleteReservationHandler}
            />
          )}
        </ul>
      ))}
    </section>
  );
};

export default ReservationsPage;
