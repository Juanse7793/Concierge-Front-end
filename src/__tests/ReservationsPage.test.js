import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import ReservationsPage from '../pages/ReservationsPage';

const mockStore = configureMockStore();
const store = mockStore({
  user: {
    user: {
      id: 1,
      reservations: [
        {
          id: 1,
          event_id: 1,
          user_id: 1,
          start: '2022-02-02',
          end: '2022-03-03',
        },
      ],
    },
  },
  events: {
    events: [
      {
        id: 1,
        name: 'name',
        location: 'location',
        image_urls: ['images'],
        price: 123,
        start_date: '2022-01-01',
        end_date: '2023-01-01',
      },
      {
        id: 2,
        name: 'bar',
        location: 'ation',
        start_date: '2022-01-01',
        end_date: '2023-01-01',
        price: 123,
        image_urls: ['images/bar.jpg'],
      },
    ],
  },
});

jest.mock('../redux/reducers/users', () => ({
  signIn: () => ({ type: 'MOCK' }),
}));

it('matches snapshot', () => {
  const component = renderer.create(
    <Provider store={store}>
      <ReservationsPage />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
