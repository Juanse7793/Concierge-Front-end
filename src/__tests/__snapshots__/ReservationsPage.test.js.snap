// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section>
  <ul>
    <li
      className="row pill list"
    >
      <p>
        name
      </p>
      <div
        className="row"
      >
        <small>
          Tue Feb 01 2022 ~ Wed Mar 02 2022
        </small>
        <button
          className="red pill link"
          id={1}
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </li>
  </ul>
</section>
`;
